DevTools/HackStudio/Git/GitRepo.cpp

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2020, Itamar S. <itamar8910@gmail.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "GitRepo.h"
+#include <AK/ScopeGuard.h>
+#include <LibCore/File.h>
+#include <fcntl.h>
+#include <spawn.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <sys/wait.h>
+
+namespace HackStudio {
+
+GitRepo::CreateResult GitRepo::try_to_create(const LexicalPath& repository_root)
+{
+    if (!git_is_installed()) {
+        return { CreateResult::Type::GitProgramNotFound, nullptr };
+    }
+    if (!git_repo_exists(repository_root)) {
+        return { CreateResult::Type::NoGitRepo, nullptr };
+    }
+
+    return { CreateResult::Type::Success, adopt(*new GitRepo(repository_root)) };
+}
+
+RefPtr<GitRepo> GitRepo::initialize_repository(const LexicalPath& repository_root)
+{
+    auto res = command_wrapper("init", repository_root);
+    if (!res.has_value())
+        return {};
+
+    ASSERT(git_repo_exists(repository_root));
+    return adopt(*new GitRepo(repository_root));
+}
+
+Vector<LexicalPath> GitRepo::unstaged_files() const
+{
+    auto modified = modified_files();
+    auto untracked = untracked_files();
+    modified.append(untracked.data(), untracked.size());
```

Comment:
I think you can do `modified.append(untracked);`
If not, we should make that work :)